src/core/qgsdataprovider.h

```diff
@@ -104,6 +104,7 @@ class CORE_EXPORT QgsDataProvider : public QObject
     struct ProviderOptions
     {
       QgsCoordinateTransformContext transformContext;
+      bool trustDatasourceConfig; //!< Trust datasource config (primary key unicity, geometry type and srid, etc). Improves provider load time.
```

Comment:
It'd be more future-friendly if we also used a new flags setup here instead of bools. (Don't forget the "since" annotation too!)

Reply:
I have removed my changes on `struct ProviderOptions` and added `ReadFlags`